### PR TITLE
Certificates Server Idempotency Issue #178

### LINKED ISSUE #178: Certificates - Idempotency issue
## Scenario/Intent

Hello,

I am adding multiple server certificates in OneView: 
image
and it works fine. However when running the tasks a second time with the same certificates I still have changes.

### main.yml

```
- name: Create a Server Certificate
  oneview_certificates_server:
    config: "{{ config }}"
    state: present
    name: "{{ item.aliasName }}"
    data:
      certificateDetails:
        - aliasName: "{{ item.aliasName }}"
          base64Data: "{{ item.base64Data }}"
  loop: "{{ certificates }}"
```

Example of input certificates:
```
    certificates:
        - aliasName: "test-certificate-2"
          base64Data: "-----BEGIN CERTIFICATE-----\n....\n-----END CERTIFICATE-----"
        - aliasName: "test-certificate"
          base64Data: "-----BEGIN CERTIFICATE-----\n....\n-----END CERTIFICATE-----"
```

## Environment Details

- OneView Ansible Collection: v6.6.0
- OneView Appliance Version: 6.10
- Ansible version: 2.9.6
- Python version: 3.8.10

## Expected Result

We have idempotency when we run the same playbook with the same certificate input data multiple times.

## Actual Result

```
TASK [certificate : Create a Server Certificate] *******************************
changed: [localhost -> localhost] => (item={'aliasName': 'test-certificate', 'base64Data':

[...]

PLAY RECAP *********************************************************************
localhost                  : ok=3    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0 
```

### PR DESCRIPTION
**Pull request by alisha-k-kalladassery**
### Description
Modifying dict_merge in oneview.py to avoid idempotency

### Issues Resolved
#178 

### Check List
- [ ] New functionality includes sanity testing.
  - [ ] All sanity tests pass. (`$ ansible-test sanity`).
  - [ ] All unit tests pass.
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in the examples (please include helpful comments).

### CODE REVIEW

File: plugins/module_utils/oneview.py [hunk 1]
```diff
@@ -105,7 +105,13 @@ def dict_merge(original_resource_dict, data_dict):
         elif isinstance(resource_dict[key], dict) and isinstance(data_dict[key], Mapping):
             resource_dict[key] = dict_merge(resource_dict[key], data_dict[key])
         elif isinstance(resource_dict[key], list) and isinstance(data_dict[key], list):
-            resource_dict[key] = data_dict[key]
+            if len(resource_dict[key]) == 1 and len(data_dict[key]) == 1:
+                if isinstance(resource_dict[key][0], dict) and isinstance(data_dict[key][0], dict):
```
Comment (AvijitOS) [inline plugins/module_utils/oneview.py:109]: Why not isinstance(data_dict[key][0], Mapping) used?

File: tests/unit/test_oneview.py [hunk 2]
```diff
@@ -2032,6 +2033,17 @@ def test_sort_by_keys(self):
                          dict(networkType="Ethernet", name="name-2")]
         assert result1 == expected_list
 
+    def test_merge_dict_with_single_element_list_inside_dict(self):
+        original_dict = dict(test1=[dict(id=2, allocatedMbps=1000, mac="E2:4B:0D:30:00:0B", requestedMbps=1000)])
+
+        dict_with_changes = dict(test1=[dict(id=2, requestedMbps=2700, allocatedVFs=3500)])
```
Comment (AvijitOS) [inline tests/unit/test_oneview.py:2039]: not seeing all other test cases like new record added/deleted in the list?
Also we should have negative test cases where first list with non empty dict and second list with empty dict
